src/controllers/__tests__/userController.test.js

```diff
@@ -0,0 +1,58 @@
+import { BAD_REQUEST, OK, CREATED,  } from 'http-status-codes';
+
+import UserController from '../userController';
+import UserService from '../../services/userService';
+
+jest.mock('../../services/userService',);
+
+describe("userController", () => {
+    let userService;
+    let userController;
+
+    const user1 = Object.freeze({id: 1});
+    const methods = [
+        { method: 'getUsers', statusCode: OK, resData: [user1, { id: 2 }] },
+        { method: 'getUserById', statusCode: OK, resData: [user1] },
+        { method: 'createUser', statusCode: CREATED, resData: user1 },
+        { method: 'updateUser', statusCode: OK, user1 },
+        { method: 'deleteUser', statusCode: OK, resData: {} }];
+
+    methods.forEach(({ method, statusCode, resData}) => {
+        describe(method, () => {
+            it(`should return ${statusCode} status with data`, async () => {
+                userService = new UserService({});
+                userService[method].mockImplementation(() => Promise.resolve(resData));
+                userController = new UserController(userService);
+    
+                const mockResponse = () => {
```

Comment:
Consider using "beforEach" functionality of Jest for the sake of not repeating this part in every test case.